osbs/api.py

```diff
@@ -10,11 +10,25 @@
 from osbs.build import BuildManager, BuildResponse
 from osbs.constants import DEFAULT_NAMESPACE
 from osbs.core import Openshift
-from osbs.exceptions import OsbsResponseException
+from osbs.exceptions import OsbsException, OsbsResponseException
 
 
+def osbsapi(func):
+    def catch_exceptions(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except OsbsException:
+            # Re-raise OsbsExceptions
+            raise
+        except Exception as e:
+            # Convert anything else to OsbsException
+            raise OsbsException(e)
```

Comment:
I would be in favor of expanding message here, because we lose information about original exc type:
```
In [2]: from osbs.exceptions import OsbsException
In [3]: a = {}
In [4]: try:
   ...:     a["a"]
   ...: except Exception as ex:
   ...:     raise OsbsException(ex)
   ...: 
---------------------------------------------------------------------------
OsbsException                             Traceback (most recent call last)
<ipython-input-4-9e9da677c841> in <module>()
      2     a["a"]
      3 except Exception as ex:
----> 4     raise OsbsException(ex)
        global OsbsException = <class 'osbs.exceptions.OsbsException'>
        global ex = KeyError('a',)
      5 

OsbsException: 'a'

In [5]: try:
    a["a"]
except Exception as ex:
    raise OsbsException(repr(ex))
   ...: 
---------------------------------------------------------------------------
OsbsException                             Traceback (most recent call last)
<ipython-input-5-fe87eae34c24> in <module>()
      2     a["a"]
      3 except Exception as ex:
----> 4     raise OsbsException(repr(ex))
        global OsbsException = <class 'osbs.exceptions.OsbsException'>
        global repr = undefined
        global ex = KeyError('a',)
      5 

OsbsException: KeyError('a',)
```
What do you think, Tim?